fortuna/src/api/revelation.rs

```diff
@@ -59,14 +60,16 @@ pub async fn revelation(
         .get(&chain_id)
         .ok_or_else(|| RestError::InvalidChainId)?;
 
-    let maybe_request = state
-        .contract
-        .get_request(state.provider_address, sequence)
-        .await
-        .map_err(|_| RestError::TemporarilyUnavailable)?;
+    let maybe_request_fut = state.contract.get_request(state.provider_address, sequence);
```

Comment:
Obviously not for this PR at all but just wanted to note the fact that the endpoint is doing contract interactions and isn't abstracted away for other chains yet, we probably don't want to be directly interacting with state in our endpoints without business logic abstracting the chain implementation.

Reply:
I’m confused by this comment

the `contract` referenced here is a `dyn EntropyReader`, which is an abstraction that is (mostly) chain agnostic. (It's chain agnostic except for that it uses ethereum addresses, but we could easily make the address fields into strings to support other chains. It's a trivial refactor, which is also the reason I haven't done it yet)